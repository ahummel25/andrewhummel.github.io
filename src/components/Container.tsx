import * as React from "react";
import styled from "@emotion/styled";

import { widths } from "../styles/variables";
import { getEmSize } from "../styles/mixins";

interface ContainerProps {
  className?: string;
}

const StyledContainer = styled.div`
  margin-left: auto;
  margin-right: auto;
  width: auto;
  max-width: ${getEmSize(widths.lg)}em;
`;

const Container: React.FC<ContainerProps> = ({
  children,
  className
}): JSX.Element => (
  <StyledContainer className={className}>{children}</StyledContainer>
);

export default Container;
