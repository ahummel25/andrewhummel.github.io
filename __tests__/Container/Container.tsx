import React from 'react';
import renderer from 'react-test-renderer';

import Container from '../../src/components/Container';

describe('Container', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Container />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional className prop', () => {
    const tree = renderer
      .create(<Container className="my-mock-class" />)
      .toTree();

    expect(tree?.props.className).toBe('my-mock-class');
  });
});
