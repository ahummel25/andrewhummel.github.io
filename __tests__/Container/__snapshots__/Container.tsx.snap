// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
.emotion-0 {
  margin-left: auto;
  margin-right: auto;
  width: auto;
  max-width: 60em;
}

.emotion-0 {
  margin-left: auto;
  margin-right: auto;
  width: auto;
  max-width: 60em;
}

<div
  className="emotion-0"
/>
`;
